### ISSUE TITLE
Impossible to use with  overv / openstreetmap-tile-server ?

### ISSUE THREAD

**Issue by Gaibhne** [giggls/openstreetmap-carto-de#60]
I was trying to set up the Dockerized OSM tileserver via https://github.com/Overv/openstreetmap-tile-server which seems pretty verbatim, but stumbled pretty early on trying to use this style with it. The first error blocking any progress is that `planet_osm_hstore_roads` does not exist in my database.

Is there a way around that ? Or is that an issue with 'the other side' of my setup ?

**Comment by giggls** [comment]
Unfortunately I do not follow upstream for some time now. Maybe the roads table has gone for good.

Unfortunately development of this for is still stuck. See here for the reason:
https://blog.geggus.net/2020/09/why-development-of-german-osm-carto-style-is-currently-stuck/

Did you have a look at https://github.com/giggls/openstreetmap-carto-de/blob/master/INSTALL-de.md

**Comment by Gaibhne** [comment]
Yes, that is what I'm struggling with - the second step expects that table to be there, which apparently has not been the case for a few years, or at least I could not find any recent mention of it.

I'm not particularly concerned with upstream, but I do need my tile server to run in Docker, and a map style without a working tile server is sadly not terribly useful for me. It seems that post is from almost two years ago, so I should probably give up on the German map ever being usable again :(

I am unsure if I understand what this repository is really for. Do people run the style on very outdated tile servers, or is there an alternative tile server that can be used with this style somewhere ?

**Comment by giggls** [comment]
Again, did you even read the [INSTALL-de.md](https://github.com/giggls/openstreetmap-carto-de/blob/master/INSTALL-de.md)?

Quote from this file:
_"The actual rendering is done using database views providing virtual columns instead of using the raw tables generated by osm2pgsql"_

I just cloned upstream carto style and it is still using planet_osm_roads so there is no incompatibility concerning a missing roads table. I supose you just did not use hstore-only.style for database import. It should still be possible to use upstream database layout. See https://github.com/giggls/openstreetmap-carto-de/tree/master/contrib/use-upstream-database

P.S.: I do not have any clue about the magic this docker thingy does.

**Comment by Gaibhne** [comment]
I did, like I said. I executed the first line as per instructions, which worked, and when I got to the second step, it failed, complaining that those tables did not exist. I will go through the steps again to make sure I did nothing wrong, but I'm pretty sure I followed all instructions to the letter.

**Comment by giggls** [comment]
Well the Instructions you are following are obviously not instructions from this repository here but from a repository I did not even know about until reading your issue today.

This is just the German carto style repo which uses a slightly different database scheme than upstream carto (which is compatible to upstream carto using views, but still) and this is not about setting up tileservers at all.

It is documented (in the already mentioned [INSTALL-de.md](https://github.com/giggls/openstreetmap-carto-de/blob/master/INSTALL-de.md)) how to run osm2pgsql so you need to change this docker thingy accordingly.

What are you expecting from me to do to resolve this "issue"?
Trying to understand the magic this docker thingy does?

Regardless of the database scheme it is likely, that the missing l10n features in upstream will also bite you.

Frankly, what I hate about those random modern container based setups. They are easy to run if you do _not_ want to change anything because you do not need to understand what's going on behind the scenes at all.
